lib/utility/runJoin.js

```diff
@@ -19,17 +19,22 @@ exports.runJoin = function runJoin(Model, isCollection, instances, models, isInn
 		var key0 = Object.keys(instances)[0],
 			instance0 = instances[key0];
 		if (isCollection) {
-			async.map(instance0,
-				function mapItems(instance, cb) {
-					self.runJoin(Model, false, {key0: instance}, models, isInnerJoin, cb);
-				}, function (err, results) {
-					if (err) {
-						next(err);
-					}
-					else {
-						next(null, _.isArray(results) ? _.compact(results) : results);
-					}
-				});
+			if (instance0) {
```

Comment:
You don't need to wrap this in an extra if-else block if you just add `if (isCollection && instance0)` in L21. That will have the same effect.